controlplane/kubeadm/internal/failure_domain_test.go

```diff
@@ -120,11 +120,11 @@ func TestNewFailureDomainPickMost(t *testing.T) {
 			expected: nil,
 		},
```

Comment:
This is missing coverage for non-failure domain cases. I think it just ignores them, but worth demonstrating that.

Reply:
Added more tests.